### PR TITLE
Change phys_cpu_num/pcpu_active_bitmap to static 

### PR DESCRIPTION
**Pull request by mingqiangchi**
hv:Change phys_cpu_num to static

-- change phys_cpu_num to static
-- add get_pcpu_nums() and is_pcpu_active() APIs
-- replace phys_cpu_num with get_pcpu_nums() except cpu.c
-- Change pcpu_active_bitmap to static, only used in   cpu.c

Tracked-On: #1842
Signed-off-by: Mingqiang Chi <mingqiang.chi@intel.com>
Acked-by: Anthony Xu <anthony.xu@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
2208<br>3 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/guest/vm.c&nbsp;&nbsp;function:prepare_vm0&nbsp;&nbsp;offset:401:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Function return value potentially unused. : vm0_desc.vm_hw_num_cores<br><br>2.filename:/hypervisor/arch/x86/cpu.c&nbsp;&nbsp;function:get_pcpu_active_bitmap&nbsp;&nbsp;offset:81:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : get_pcpu_active_bitmap<br><br>3.filename:/hypervisor/arch/x86/cpu.c&nbsp;&nbsp;function:start_cpu&nbsp;&nbsp;offset:237:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Logical conjunctions need brackets.<br><br>

File: hypervisor/arch/x86/cpu.c [hunk 1]
```diff
@@ -77,6 +77,12 @@ bool is_pcpu_active(uint16_t pcpu_id)
 {
 	return bitmap_test(pcpu_id, &pcpu_active_bitmap);
 }
+
+uint64_t get_pcpu_active_bitmap(void)
+{
```
Comment (dongyaozu) [inline hypervisor/arch/x86/cpu.c]: From English p.o.v., This should be get_active_pcpu_bitmap...

**Comment by mingqiangchi** [comment]
checked with Shiqing, 3 violations are false positive

File: hypervisor/arch/x86/cpu.c (hunk 1 above)
Comment (mingqiangchi) [reply hypervisor/arch/x86/cpu.c]: ok, will update

**Comment by HuiHuangShi** [comment]
2208<br>3 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/guest/vm.c&nbsp;&nbsp;function:prepare_vm0&nbsp;&nbsp;offset:401:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Function return value potentially unused. : vm0_desc.vm_hw_num_cores<br><br>2.filename:/hypervisor/arch/x86/cpu.c&nbsp;&nbsp;function:get_active_pcpu_bitmap&nbsp;&nbsp;offset:81:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : get_active_pcpu_bitmap<br><br>3.filename:/hypervisor/arch/x86/cpu.c&nbsp;&nbsp;function:start_cpu&nbsp;&nbsp;offset:237:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Logical conjunctions need brackets.<br><br>

**Comment by HuiHuangShi** [comment]
2208<br>3 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/guest/vm.c&nbsp;&nbsp;function:prepare_vm0&nbsp;&nbsp;offset:401:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Function return value potentially unused. : vm0_desc.vm_hw_num_cores<br><br>2.filename:/hypervisor/arch/x86/cpu.c&nbsp;&nbsp;function:get_active_pcpu_bitmap&nbsp;&nbsp;offset:81:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : get_active_pcpu_bitmap<br><br>3.filename:/hypervisor/arch/x86/cpu.c&nbsp;&nbsp;function:start_cpu&nbsp;&nbsp;offset:237:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Logical conjunctions need brackets.<br><br>

**Comment by wenlingz** [comment]
OK to verify

**Comment by HuiHuangShi** [comment]
2208<br>3 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/guest/vm.c&nbsp;&nbsp;function:prepare_vm0&nbsp;&nbsp;offset:401:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Function return value potentially unused. : vm0_desc.vm_hw_num_cores<br><br>2.filename:/hypervisor/arch/x86/cpu.c&nbsp;&nbsp;function:get_active_pcpu_bitmap&nbsp;&nbsp;offset:81:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure is not called or referenced in code analysed. : get_active_pcpu_bitmap<br><br>3.filename:/hypervisor/arch/x86/cpu.c&nbsp;&nbsp;function:start_cpu&nbsp;&nbsp;offset:237:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Logical conjunctions need brackets.<br><br>

**Pull request merged**